lib/_007/Val.pm

```diff
@@ -205,10 +205,86 @@ class Val::Str does Val {
 ###     say(/"Bond"/.search("J. Bond"));        # --> `True`
 ###
 class Val::Regex does Val {
-    has Val::Str $.contents;
+    # note: a regex should probably keep its lexpad or something to resolve calls&identifiers
```

Comment:
Or surrounding frame. This is going to become an issue fast, so we'd better have a plan/design there.

Reply:
As a first approximation: a regex is a first-class value that can be invoked, just as much as a sub is. Therefore it needs to have an `outer-frame` property &mdash; otherwise lookups are going to go to the wrong place when the regex runs after having been passed to another frame.

Or, hm. If we don't plan to declare things inside the regex, then we technically don't have to `enter` and `leave` it, and then `outer-frame` can be just `frame`.